content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,242 @@
+
+Read on to discover everything that shipped in Sourcegraph 4.0.  
+
+### 🧠 Code intelligence: uplevel your code search
+
+<br />
+<a href="/code-insights"><Badge text="Code Insights" color="green" size="small" /></a>
+
+#### Understand usage and search structure with Code Insights dashboards within the search UI
+[Code Insights](https://about.sourcegraph.com/code-insights) turns your code into a queryable database to create customizable, visual dashboards. We're bringing the power of Code Insights directly into the search UI to surface high-level information about your code, help you answer questions about your code, and refine your search results.  
+
+Search aggregation charts appear on the search results page to help you answer questions, like:
+- How many different versions of a library or package are present in your code, and which is most common? 
+- Which files have the majority of these search results?
+- Which repositories are this library used in most?
+- Which library components are used most? By who?
+- What are all the possible arguments we pass to this function? 
+
+You can choose to group your search results by location (repo or file), author, or arbitrary capture group pattern. Here are some [example search aggregations to get started with.](https://docs.sourcegraph.com/code_insights/references/search_aggregations_use_cases)
+
+<Video 
+  source={{
+    webm: 'blog/release-post/4.0/Search_Aggregations_4.0',
+    mp4: 'blog/release-post/4.0/Search_Aggregations_4.0'
+  }}
+  loop={true}
+  title="High-level aggregations"
+  caption="The power of Code Insights is now within the search UI to surface usage patterns about your code"
+/>
+<a href="https://docs.sourcegraph.com/code_insights/explanations/search_results_aggregations" className="tw-not-italic tw-flex tw-items-center tw-mb-sm">Docs<OpenInNewIcon className="tw-ml-xxs" size={18} /></a>
+  
+<br />
+<a href="/code-search"><Badge text="Code Search" color="cerise" size="small" /></a>
+
+#### A faster, simpler search experience 
+We streamlined and simplified the search user interface to make the platform more intuitive. The search homepage now has contextual tips and recommendations using real data from your instance, including repository names, file names, and authors. This information makes it easier and faster than ever to run a successful search query. 
+
+Search results are now front and center with the improved interface. The search sidebar has moved to the right side of the screen, and we eliminated several non-essential elements to ensure the information you need is easy to access. Filters make it effortless for you to answer questions about your code, and the sidebar can be collapsed altogether to reduce noise. 
+
+Major performance improvements make rendering search results, files, blobs, and repository pages faster than ever.
```

Comment:
very minor: This paragraph looks a bit awkward being much shorter than the previous two and in a slightly different tone. But I'm not sure how much detail we want to add here without getting too technical. I've tried rewording it a little bit to fit in more with the written style of the previous paragraphs. Files and blobs are the same thing so no need to mention both.

```suggestion
We also made major performance improvements to make rendering search results, files, and repositories faster than ever. 
```